app/src/main/java/com/jerboa/ui/components/post/PostViewModel.kt

```diff
@@ -1,284 +1,315 @@
 
 const val COMMENTS_DEPTH_MAX = 6
 
-typealias PostId = Int
-typealias CommentId = Int
-
 class PostViewModel : ViewModel() {
 
-    var postView = mutableStateOf<PostView?>(null)
+    var postRes: ApiState<GetPostResponse> by mutableStateOf(ApiState.Empty)
         private set
 
-    // If this is set, its a comment type view
-    var commentId = mutableStateOf<Int?>(null)
-        private set
-    var comments = mutableStateListOf<CommentView>()
+    var commentsRes: ApiState<GetCommentsResponse> by mutableStateOf(ApiState.Empty)
         private set
-    var commentTree = mutableStateListOf<CommentNodeData>()
-        private set
-    var moderators = mutableStateListOf<CommunityModeratorView>()
-        private set
-    var loading: Boolean by mutableStateOf(false)
+
+    // If this is set, its a comment type view
+    var id by mutableStateOf<Either<PostId, CommentId>?>(null)
         private set
-    var sortType = mutableStateOf(CommentSortType.Hot)
+    var sortType by mutableStateOf(CommentSortType.Hot)
         private set
 
-    fun fetchPost(
-        id: Either<PostId, CommentId>,
-        clearPost: Boolean = false,
-        clearComments: Boolean = false,
-        account: Account?,
-        ctx: Context,
-        changeSortType: CommentSortType? = null,
-    ) {
-        val api = API.getInstance()
+    private var fetchingMore by mutableStateOf(false)
 
-        // Set the commentId for the right case
-        id.fold({ commentId.value = null }, { commentId.value = it })
+    private var likeCommentRes: ApiState<CommentResponse> by mutableStateOf(ApiState.Empty)
+    private var saveCommentRes: ApiState<CommentResponse> by mutableStateOf(ApiState.Empty)
+    private var deleteCommentRes: ApiState<CommentResponse> by mutableStateOf(ApiState.Empty)
 
-        viewModelScope.launch {
-            try {
-                Log.d(
-                    "jerboa",
-                    "Fetching post: $id",
-                )
-
-                if (clearPost) {
-                    postView.value = null
-                }
+    private var likePostRes: ApiState<PostResponse> by mutableStateOf(ApiState.Empty)
+    private var savePostRes: ApiState<PostResponse> by mutableStateOf(ApiState.Empty)
+    private var deletePostRes: ApiState<PostResponse> by mutableStateOf(ApiState.Empty)
+    private var blockCommunityRes: ApiState<BlockCommunityResponse> by
+        mutableStateOf(ApiState.Empty)
+    private var blockPersonRes: ApiState<BlockPersonResponse> by mutableStateOf(ApiState.Empty)
 
-                if (clearComments) {
-                    comments.clear()
-                    commentTree.clear()
-                }
+    fun initialize(
+        id: Either<PostId, CommentId>,
+    ) {
+        this.id = id
+    }
 
-                changeSortType?.also {
-                    sortType.value = it
-                }
+    fun updateSortType(sortType: CommentSortType) {
+        this.sortType = sortType
+    }
```

Comment:
Should the setter above just be made public?

Reply:
That function is public, its called by MainActivity.